website/servers/manage.php

```diff
@@ -0,0 +1,344 @@
+	if (count($failures) > 0) {
+		echo ("(" . count($failures) . "user (s) could not be added: " . implode(", ", $failures) . ")<br />");
+	}
+}
+
+echo('<br />');
+if (isset($serverinfo['blacklist']) && count($serverinfo['blacklist'])) {
+?>
+	<hr />
+	<form action="/servers/manage.php" data-target="push">
+		<label>(Separate userids with commas)</label><br />
+		<input type="text" name="unbanowners" class="textbox" /><br /><br />
+		<button type="submit">Remove users from list</button>
+	</form>
+	<?php
+}
+
+if (isset($_REQUEST['unbanowners']) && isset($serverinfo['blacklist'])) {
+	$unbans = explode(',', $_REQUEST['unbanowners']);
+	if (count($unbans)) {
+		$successes = [];
+		$failures = [];
+		foreach ($unbans as $userid) {
+			$userid = $users->userid($userid);
+			$idx = array_search($userid, $serverinfo['blacklist']);
+			if ($idx) {
+				$successes[] = $userid;
+				array_splice($serverinfo['blacklist'], $idx);
+			} else {
+				$failures[] = $userid;
+			}
+		}
+	}
+	if (count($successes)) {
+		echo(
+			"<strong>" . count($successes) .
+			" user (s) were removed from the blacklist.</strong>" .
+			"<label>(" . implode(", ", $successes) . ")</label><br />"
+		);
+	}
+	if (count($failures)) {
+		echo(
+			"<strong style=\"color:red\">" . count($failures) .
+			"user (s) were not on the blacklist. </strong>" .
+			" <label>(" . implode(', ', $failures) . ")</label><br />"
+		);
+	}
+}
+?>
+	</div>
+	</td> <td style="vertical-align: top; max-width: 300px">
+	<form action="/servers/manage.php" data-target="push">
+		<label>(Separate hosts with commas)</label><br />
+		<input type="text" name="banhosts" class="textbox" /><br /><br />
+		<button type="submit">Add hosts to list</button>
+	</form>
+	</div>
+<?php
+
+if (isset($_REQUEST['unbanhosts']) && mb_strlen($_REQUEST['unbanhosts'])) {
+	$hosts = explode(',', $_REQUEST['unbanhosts']);
+	$successes = [];
+	$failures = [];
+	foreach ($hosts as $i=>$host) {
+		$host = trim($host);
+		if (!mb_strlen($host)) continue;
+		if (!isset($serverinfo['bannedhosts'])) {
+			$serverinfo['bannedhosts'] = [];
+		}
+		if (!in_array($host, $serverinfo['bannedhosts'])) {
+			$failures[] = $host;
+		} else {
+			$successes[] = $host;
+			array_splice($serverinfo['bannedhosts'], $i);
+		}
+	}
+	writeserverinfo();
+	$scount = count($successes);
+	$fcount = count($failures);
+	echo('Removed ' . $scount . " host (s) from the blacklist. (" . implode(', ', $successes) . ")");
+	if ($fcount > 0) {
+		echo("Failed to remove " . $fcount . " host (s) from the blacklist. (" . implode(', ', $failures) . ")");
+	}
+	echo('</div>');
+}
+
+?>
+	<br /><hr />
+	<form action="/servers/manage.php" data-target="push">
+		<label>(Separate hosts with commas)</label><br />
+		<input type="text" name="unbanhosts" class="textbox" /><br /><br />
+		<button type="submit">Remove hosts from list</button>
+	</form>
+<?php
+
+echo('</td></tr>');
+
+echo('</table></div></div>');
+
+echo '<center>';
```

Comment:
This is so confusing...

```suggestion
</td></tr>
</table></div></div>
<center>
<?php
```